extension/jaegerremotesampling/extension_test.go

```diff
@@ -16,89 +16,79 @@ package jaegerremotesampling
 
 import (
 	"context"
-	"errors"
+	"fmt"
+	"net"
+	"path/filepath"
 	"testing"
 
+	"github.com/jaegertracing/jaeger/proto-gen/api_v2"
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
-	"go.opentelemetry.io/collector/component"
 	"go.opentelemetry.io/collector/component/componenttest"
+	"go.opentelemetry.io/collector/config/configgrpc"
+	"google.golang.org/grpc"
 )
 
 func TestNewExtension(t *testing.T) {
 	// test
-	e, err := newExtension(createDefaultConfig().(*Config), componenttest.NewNopTelemetrySettings())
-	require.NoError(t, err)
+	cfg := createDefaultConfig().(*Config)
+	cfg.Source.File = filepath.Join("testdata", "strategy.json")
+	e := newExtension(cfg, componenttest.NewNopTelemetrySettings())
 
 	// verify
 	assert.NotNil(t, e)
 }
 
-func TestStartAndShutdown(t *testing.T) {
+func TestStartAndShutdownLocalFile(t *testing.T) {
 	// prepare
-	e, err := newExtension(createDefaultConfig().(*Config), componenttest.NewNopTelemetrySettings())
+	cfg := createDefaultConfig().(*Config)
+	cfg.Source.File = filepath.Join("testdata", "strategy.json")
+
+	e := newExtension(cfg, componenttest.NewNopTelemetrySettings())
 	require.NotNil(t, e)
-	require.NoError(t, err)
 	require.NoError(t, e.Start(context.Background(), componenttest.NewNopHost()))
 
 	// test and verify
 	assert.NoError(t, e.Shutdown(context.Background()))
 }
 
-func TestFailedToStartHTTPServer(t *testing.T) {
```

Comment:
Is this test just not needed anymore?

Reply:
Kind of: previously, I had the ability to provide an HTTP server, making it trivial to test a failure scenario. Now, I moved the entire initialization logic to the start function, so that an HTTP server is created there. Failure testing is now more complicated.